datafusion/physical-plan/src/execution_plan.rs

```diff
@@ -743,6 +733,38 @@ pub enum EmissionType {
     Both,
 }
 
+/// Represents whether an operator's `Stream` has been implemented to actively cooperate with the
+/// Tokio scheduler or not.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum SchedulingType {
+    /// The stream generated by [`execute`](ExecutionPlan::execute) does not actively participate in
+    /// cooperative scheduling. This means the implementation of the `Stream` returned by
+    /// [`ExecutionPlan::execute`] does not contain explicit cooperative yield points.
```

Comment:
I think it would also be helpful here to give an explicitl example of what a cooperative yield point is. For example
```suggestion
    /// [`ExecutionPlan::execute`] does not contain explicit cooperative yield points such as 
    /// `await` or [`tokio::task::yield_now`].
```

Reply:
I think it would also help to document here what the implications of returning one vs the other.

I think the implication is that if an `ExecutionPlan` returns `Blocking` that DataFusion's planner will automatically wrap the stream in `make_cooperative` for example

Reply:
I kind of intentionally omitted that since it's describing the implementation a bit too much. Trying to keep this as declarative as possible.

I renamed `Blocking` to `NonCooperative` which makes it sound less like a bad thing. `NonCooperative` doesn't necessarily mean `make_cooperative` wrappers will get inserted. That's only done for leave nodes and exchanges.